docs(scheduler):resource grouping documentation added (#363)
telerik/blazor-docs: components/scheduler/resource-grouping.md

---
title: Resource Grouping
page_title: Scheduler - Resource Grouping

> description: Resource Grouping in the Scheduler for Blazor.

Reviewer: Use different keywords or different grammar forms:

```suggestion
description: Group Resources in the Scheduler for Blazor.
```

slug: scheduler-resource-grouping
tags: telerik,blazor,scheduler,resource,grouping
published: true
position: 33
---

# Scheduler Resource Grouping

> The feature allows grouping Scheduler views by one or more resources. All available supported views: `Day`, `Week`, `MultiDay` and `Month` can render both `horizontal` and `vertical` grouping.

Reviewer: We group events, not views, right?

```suggestion
The feature allows grouping Scheduler events by one or more resources. All available supported views: `Day`, `Week`, `MultiDay` and `Month` can render both `horizontal` and `vertical` grouping.
```

This article contains the following sections:

* [Basics](#basics)
    * [Define Settings](#define-settings)
* [Examples](#examples)
    * [Resource Grouping by one resource](#resource-grouping-by-one-resource)
    * [Resource Grouping by multiple resources](#resource-grouping-by-multiple-resources)

## Basics

In the grouping rendering in both horizontal and vertical orientation, the view tables are rendered next to each other. 

Moving an appointment between the resource tables is allowed, and upon dropping, the appointment resource changes alongside the start date.

### Define Settings

> To configure the group rendering, define a dedicated settings tag, called a `SchedulerGroupSettings` inside the `SchedulerSettings` tag.

Reviewer: Removing self-explanatory details.

```suggestion
To configure the group rendering, define the `SchedulerGroupSettings` tag inside the `SchedulerSettings` tag.
```

The settings tag will have the following Parameters:

> * `Resources(List<string>)` - provides a list of one or more resource names, which will be used to group the scheduler by.

Reviewer: ```suggestion
* `Resources(List<string>)` - provides a list of one or more resource names, which will be used to group the events.
```

> * `Orientation(SchedulerGroupOrientation)` - has two values: `Horizontal(default)` and `Vertical`. Determines the direction in which the resource tables are rendered.

Reviewer: ```suggestion
* `Orientation(SchedulerGroupOrientation)` - has two values: `Horizontal` (default) and `Vertical`. Determines the direction in which the resource tables are rendered.
```
